lib/ramble/ramble/software_environments.py

```diff
@@ -466,14 +470,24 @@ def __eq__(self, other):
         Returns:
             (bool): True if environments are equivalent, False otherwise
         """
-        equal = self.name == other.name and len(self._packages) == len(other._packages)
+        equal = self.name == other.name
 
         if not equal:
             return False
```

Comment:
nit: is there much benefit to splitting this calculation out? 

```suggestion
        if not self.name == other.name:
            return False
```

Reply:
No, it used to be more complex but we removed the other part of it. I've changed it to what you suggested.